atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ProfilingKeyValueService.java

```diff
@@ -189,20 +220,21 @@ void registerException(Exception ex) {
 
         void log() {
             stopwatch.stop();
-            Consumer<LoggingFunction> logger = (loggingMethod) -> {
+            Consumer<FlushableLoggingFunction> logger = (loggingMethod) -> {
                 primaryLogger.accept(loggingMethod, stopwatch);
                 if (result != null) {
                     additionalLoggerWithAccessToResult.accept(loggingMethod, result);
                 } else if (exception != null) {
                     loggingMethod.log("This operation has thrown an exception {}", exception);
                 }
+                loggingMethod.flush();
```

Comment:
should either make the logging method AutoCloseable and use try-with-resources or flush in a finally block

Reply:
yeah, AutoCloseable (and try with resources) sounds good here!